yaml/step.go

```diff
@@ -74,6 +77,32 @@ func (s *StepSlice) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return err
 	}
 
+	// iterate through each element in the step slice
+	for _, step := range *stepSlice {
+		// implicitly set `pull` field if empty
+		if len(step.Pull) == 0 {
+			step.Pull = constants.PullNotPresent
+		}
```

Comment:
Set a default pull policy of `not_present` for the `github.com/go-vela/types/yaml.Step` type